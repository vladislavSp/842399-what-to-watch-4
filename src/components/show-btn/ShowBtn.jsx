import React from 'react';
import PropTypes from 'prop-types';

export const ShowBtn = ({filmLength, onViewBtnClick}) => {
  return <div className="catalog__more">
    <button
      onClick={() => onViewBtnClick(filmLength)}
      className="catalog__button"
      type="button">
      Show more
    </button>
  </div>;
};

ShowBtn.propTypes = {
  filmLength: PropTypes.number.isRequired,
  onViewBtnClick: PropTypes.func.isRequired,
};

export default ShowBtn;
