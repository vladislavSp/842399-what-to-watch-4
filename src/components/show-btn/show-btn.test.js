import React from 'react';
import renderer from 'react-test-renderer';
import ShowBtn from './show-btn.jsx';

const filmLength = 8;

it(`ShowBtn renders correctly`, () => {
  const tree = renderer.create(<ShowBtn
    filmLength={filmLength}
    onViewBtnClick={() => {}} />, {
    createNodeMock: () => {
      return {};
    }
  })
  .toJSON();
  expect(tree).toMatchSnapshot();
});
