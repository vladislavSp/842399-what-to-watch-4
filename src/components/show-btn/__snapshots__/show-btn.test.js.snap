// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowBtn renders correctly 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[Function]}
    type="button"
  >
    Show more
  </button>
</div>
`;
